### PR TITLE
Replace sagas with rtk query: SimpleMode

### PR DESCRIPTION
**Pull request by gshilin**
Starting from SimpleMode.
As a result of other modules to be outside of RTK2, we need to add a number of calls to invalidate SimpleMode due to language (UI and/or content), like here:
https://github.com/Bnei-Baruch/kmedia-mdb/pull/1120/files#diff-3e31b33783d74ed8f6ca772791095aa332a608e881dbb16a00780a23e5181958R178

### CODE REVIEW

File: server/renderer.js [hunk 1]
```diff
@@ -151,6 +175,7 @@ async function serverRenderAuthorised(req, res, next, htmlData, uiLang, bot) {
     // Dispatching languages change updates tags and sources.
     store.dispatch(settings.setUILanguage({ uiLang: cookieUILang }));
     store.dispatch(settings.setContentLanguages({ contentLanguages: cookieContentLanguages }));
+    store.dispatch(backendApi.util.invalidateTags(wholeSimpleMode));
```
Comment (bbfsdev) [inline server/renderer.js:178]: What does invalidateTags do?
whileSimpleMode is an array with one string...

File: src/components/Sections/SimpleMode/Container.js [hunk 2]
```diff
@@ -34,17 +32,10 @@ const SimpleModeContainer = () => {
     return newDate;
   });
 
-  const dispatch = useDispatch();
-
   useEffect(() => {
     setFilesLanguages(contentLanguages);
   }, [contentLanguages]);
 
-  useEffect(() => {
```
Comment (bbfsdev) [inline src/components/Sections/SimpleMode/Container.js:43]: Why we don't need this any more?

File: src/components/Sections/SimpleMode/RenderListHelpers.js [hunk 3]
```diff
@@ -130,37 +149,59 @@ const unitDerivedFiles = (unit, type, keyFilter, mimeFilter) => {
...
 
-      return lang;
-    }
+  return 0;
+};
 
-    return 0;
-  }).filter((file, index, files) => index === 0 || sortMediaFiles(files[index-1], files[index]) !== 0)
+// TODO: bbdev REFACTOR THIS TO COMMON LIBRARY
+// Order files by types and languages; remove some duplicates
```
Comment (bbfsdev) [inline src/components/Sections/SimpleMode/RenderListHelpers.js:177]: Is this comment still true?

File: src/index.js [hunk 4]
```diff
@@ -38,6 +39,10 @@ function hydrateApp(kcInfo) {
 
   const store = createStore({ ...window.__data, auth: kcInfo }, history);
 
+  // optional, but required for refetchOnFocus/refetchOnReconnect behaviors
+  // see `setupListeners` docs - takes an optional callback as the 2nd arg for customization
+  setupListeners(store.dispatch);
```
Comment (bbfsdev) [inline src/index.js:44]: Does this influence other issues not only simple-mode?

File: src/redux/api/backendApi.js [hunk 5]
```diff
@@ -0,0 +1,13 @@
+import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react';
+import { API_BACKEND } from '../../helpers/Api';
+
+export const backendApi = createApi({
```
Comment (bbfsdev) [inline src/redux/api/backendApi.js:4]: Can we cache calls here?

File: src/redux/api/backendApi.js [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+  reducerPath      : 'backendApi',
+  keepUnusedDataFor: 10,
```
Comment (bbfsdev) [inline src/redux/api/backendApi.js:6]: 10 seconds?

File: src/redux/api/simpleMode.js [hunk 7]
```diff
@@ -0,0 +1,27 @@
+import { backendApi } from './backendApi';
+import moment from 'moment/moment';
+import { Requests } from '../../helpers/Api';
+
+export const wholeSimpleMode = ['SimpleMode'];
+
+export const simpleModeApi = backendApi.injectEndpoints({
```
Comment (bbfsdev) [inline src/redux/api/simpleMode.js:7]: Did you move only simple mode to RTK2? Or everything?

File: src/redux/modules/mdb.js [hunk 8]
```diff
@@ -490,7 +491,15 @@ const mdbSlice = createSlice({
   extraReducers: builder => {
     builder
       .addCase(ssrActions.prepare, onSSRPrepare)
-      .addCase(settingsActions.setContentLanguages, () => freshStore());
+      .addCase(settingsActions.setContentLanguages, () => freshStore())
+      .addMatcher(
```
Comment (bbfsdev) [inline src/redux/modules/mdb.js:495]: Not sure what this part is for...

**Review by bbfsdev — APPROVED** [review]
Cool change! Looks good.

File: server/renderer.js (hunk 1 above)
Comment (gshilin) [reply server/renderer.js:178]: https://redux-toolkit.js.org/rtk-query/usage/automated-refetching

File: src/components/Sections/SimpleMode/Container.js (hunk 2 above)
Comment (gshilin) [reply src/components/Sections/SimpleMode/Container.js:43]: RTK handles cache, so as far as parameters are the same - the result is the same. You don't need to react on parameter change - RTK will do it by itself

File: src/components/Sections/SimpleMode/RenderListHelpers.js (hunk 3 above)
Comment (gshilin) [reply src/components/Sections/SimpleMode/RenderListHelpers.js:177]: It looks like yes, but you're welcome to suggest better wording.

File: src/index.js (hunk 4 above)
Comment (gshilin) [reply src/index.js:44]: No. It doesn't. It doesn't even influence simple-mode as it doesn't use refetchOnFocus/refetchOnReconnect behaviors. It is here for "not to forget when will start using it", but setup will be far in the past, so nobody will remember.

File: src/redux/api/backendApi.js (hunk 5 above)
Comment (gshilin) [reply src/redux/api/backendApi.js:4]: All calls are cached by RTK automatically.

File: src/redux/api/backendApi.js (hunk 6 above)
Comment (gshilin) [reply src/redux/api/backendApi.js:6]: If user will click the "Back" button in 10 sec (or will return to prev parameters in any other way) - the result will still be in cache

File: src/redux/api/simpleMode.js (hunk 7 above)
Comment (gshilin) [reply src/redux/api/simpleMode.js:7]: Only simple mode. Step-by-step.

File: src/redux/modules/mdb.js (hunk 8 above)
Comment (gshilin) [reply src/redux/modules/mdb.js:495]: I think it would be right to read a bit about RTK :)

File: src/redux/api/backendApi.js (hunk 6 above)
Comment (bbfsdev) [reply src/redux/api/backendApi.js:6]: Just comment the number with // seconds

File: src/redux/api/simpleMode.js (hunk 7 above)
Comment (bbfsdev) [reply src/redux/api/simpleMode.js:7]: V

File: src/components/Layout/HandleLanguages.js [hunk 9]
```diff
@@ -35,18 +37,21 @@ const HandleLanguages = ({ t }) => {
   const uiLanguageSelected = language => {
     updateHtmlLang(language);
     dispatch(actions.setUILanguage({ uiLang: language }));
+    dispatch(backendApi.util.invalidateTags(wholeSimpleMode));
```
Comment (davgur) [inline src/components/Layout/HandleLanguages.js:40]: why need invalidateTags here? 
languages are parts of query

**Pull request merged**